human-viz-bridge/src/iCubEffortBridge.cpp

```diff
@@ -0,0 +1,340 @@
+            effort.publisher->write();
+        }
+        
+        return true;
+    }
+    // Configure function. 
+    bool configure(ResourceFinder &rf)
+    {
+        m_robotConfigured = false;
+        
+        string moduleName = rf.check("name", Value("icub-effort-bridge"), "Checking module name").asString();
+        string urdfModelFile = rf.findFile("robotModelFilename");
+        setName(moduleName.c_str());
+        
+        //Robot model
+        const string robotModelFilename = rf.findFile("robotModelFilename");
+
+        vector<string> robot_jointList;
+        if (!parseStringListOption(rf.find("robot_JointList"), robot_jointList))
+        {
+            yError("Error while parsing the robot joint list");
+            return false;
+        }
+
+        iDynTree::ModelLoader modelLoader;
+              
+        if(!modelLoader.loadReducedModelFromFile(robotModelFilename, robot_jointList))
+        {
+            yError("Something wrong with the robot model loading!");
+            return false;
+        }
+    
+        iDynTree::Model robotModel = modelLoader.model(); 
+        iDynTree::Traversal traversal;
+        robotModel.computeFullTreeTraversal(traversal, robotModel.getDefaultBaseLink());
+        m_robotConfiguration.resize(robot_jointList.size());
+        m_robotConfiguration.zero();
+
+        size_t effortIndex = 0;
+        size_t DoFiterator = 1;
+        string robotJoint = "*";
+        vector<string> jointDegrees = {"_pitch", "_roll", "_yaw"};
+        for (size_t jointIndex = 0; jointIndex < robot_jointList.size(); jointIndex++) {
+            
+            effortsList.resize(effortIndex+1);
+            effortsList[effortIndex].jointEffort.nrDoFs = 0;
+            DoFiterator = 0;
+            
+            if (!(robot_jointList[jointIndex].find(robotJoint)!=string::npos)) {
+               if(robot_jointList[jointIndex].find(jointDegrees[0])!=string::npos || robot_jointList[jointIndex].find(jointDegrees[1])!=string::npos || robot_jointList[jointIndex].find(jointDegrees[2])!=string::npos ) 
+               {
+                   robotJoint = robot_jointList[jointIndex];
+                   size_t found = robotJoint.find("_");
+                   size_t erasePos = robotJoint.find("_",(found+1));
+                   cout << robotJoint << endl;
+                   if (erasePos < 100) {
+                       effortsList[effortIndex].jointEffort.jointName = robotJoint.erase(erasePos);
+                   } else {
+                       effortsList[effortIndex].jointEffort.jointName = robotJoint.erase(found);
+                   }
+                   effortsList[effortIndex].linkEffort = robotModel.getLinkName(traversal.getChildLinkIndexFromJointIndex(robotModel, robotModel.getJointIndex(robot_jointList[jointIndex])));
+                   bool sameJoint = robot_jointList[(jointIndex+(DoFiterator))].find(effortsList[effortIndex].jointEffort.jointName)!=string::npos; 
+                   while(sameJoint){
```

Comment:
I think this is extremely complicated! 
Furthermore, what happen if somebody uses a list like
`torso_yaw, r_elbow, torso_roll` ?